datadog/resource_datadog_dashboard.go

```diff
@@ -0,0 +1,3856 @@
+package datadog
+
+import (
+	"fmt"
+
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogDashboardCreate,
+		Update: resourceDatadogDashboardUpdate,
+		Read:   resourceDatadogDashboardRead,
+		Delete: resourceDatadogDashboardDelete,
+		Exists: resourceDatadogDashboardExists,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogDashboardImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"title": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The title of the dashboard.",
+			},
+			"widget": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The list of widgets to display on the dashboard.",
+				Elem: &schema.Resource{
+					Schema: getWidgetSchema(),
+				},
+			},
+			"layout_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The layout type of the dashboard, either 'free' or 'ordered'.",
```

Comment:
I was relying on the upstream API to handle validation in some cases to simplify the resource code, but this could be a nice candidate for `ValidateFunc`. Updated.